app.rb

```diff
@@ -1,150 +1,45 @@
-require './person'
-require './rental'
-require 'date'
-require './classroom'
-require './teacher'
-require './student'
-require './book'
+require './create_person'
+require './create_book'
+require './create_rental'
 
 class App
   def initialize
-    @rentals = []
-    @people = []
-    @books = []
-    @teachers = []
-    @students = []
-  end
-
-  def list_books
-    if @books.length.positive?
-      @books.each_with_index do |book, i|
-        p " book number: #{i}, ID: #{book.id}, Title: #{book.title}, Author: #{book.author}"
-      end
-    else
-      p 'books not found'
-    end
-  end
-
-  def list_people
-    if @people.length.positive?
-      @people.each_with_index do |person, i|
-        p "person number: #{i}, ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
-      end
-    else
-      p 'persons not found'
+    @new_person = CreatePerson.new
+    @new_book = CreateBook.new
+    @new_rental = CreateRental.new({ persons: @new_person, books: @new_book })
+    @output = 0
+  end
+
+  def options
+    print 'Welcome to School Library App!'
+    while @output != 7
+      print "\nPlease choose an option by entering a number: \n\n"
+      print "1 - List all books\n"
+      print "2 - List all people\n"
+      print "3 - Create a person\n"
+      print "4 - Create a book\n"
+      print "5 - Create a rental\n"
+      print "6 - List all rentals for a given person id\n"
+      print "7 - Exit\n"
+      start_app
     end
   end
 
-  def rent_list_by_id
-    if @rentals.length.positive?
-      p('select Person ID in the List below')
-      list_people
-      p 'Type person ID:'
-      id = gets.chomp
-
-      @rentals.select do |rent|
-        if rent.person.id == id.to_i
-          p "Date: #{rent.date}, Book '#{rent.book.title}', Author #{rent.book.author}"
-        else
-          p 'Rental not found for the selected person '
-        end
-      end
-    else
-      p 'Rental not found'
-    end
-  end
-
-  def create_person
-    p 'For student type (1), for teacher tyep (2)'
-    input = gets.chomp
-    case input.to_i
+  def start_app
+    @output = gets.chomp.to_i
+    case @output
     when 1
-      create_student_option
+      @new_book.list_books
     when 2
-      create_teacher_option
-    else
-      p 'invalid input!!! For student type (1), for teacher tyep (2)'
+      @new_person.list_people
+    when 3
+      @new_person.create_person
+    when 4
+      @new_book.create_new_book
+    when 5
+      @new_rental.create_new_rental
+    when 6
+      @new_rental.rent_list_by_id
```

Comment:
- [ ] Kindly make sure all requests for input and input validation should be placed in the main.rb file also move classes/methods whose responsibilities are to request or validate input into the main.rb, because the role of `main.rb` is  to start your code and ask the user for the initial choice of options (or to quit) 👍🏿